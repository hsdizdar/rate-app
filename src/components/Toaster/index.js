import React from "react";
import styled from "styled-components";

const ToastContainer = styled.div`
  position: absolute;
  top: 10px;
  z-index: 9999;
  height: 60px;
  width: 550px;
  background-color: #ddf9d3;
  border: 1px solid #3ca420;
  border-radius: 5px;
  opacity: 0.9;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;
const ToastImportantText = styled.span`
  color: #5aad5a;
  text-align: center;
  font-size: 21px;
  font-weight: 600;
  padding: 5px;
`;
const ToastText = styled.span`
  color: #5aad5a;
  text-align: center;
  font-size: 21px;
  padding: 5px;
`;

const Toaster = ({ isShow, linkName, message }) => {
  return (
    <ToastContainer
      style={{ display: isShow ? "flex" : "none" }}
      isShow={isShow}
    >
      <ToastImportantText>{linkName}</ToastImportantText>
      <ToastText>{message}</ToastText>
    </ToastContainer>
  );
};

export default Toaster;
