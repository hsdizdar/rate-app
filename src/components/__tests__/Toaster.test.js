import { shallow } from "enzyme";
import Toaster from "../Toaster";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const props = {
  isShow: true,
  linkName: "Hepsiburada",
  message: "added",
};

let component;

beforeEach(() => {
  component = shallow(<Toaster {...props} />);
});

describe("Toaster rendering", () => {
  it("renders correctly", () => expect(component).toMatchSnapshot());

  it("props works correctly", () => {
    expect(component.props().isShow).toEqual(props.isShow);
    expect(component.props().children[0].props.children).toEqual(
      props.linkName
    );
    expect(component.props().children[1].props.children).toEqual(props.message);
  });
});
