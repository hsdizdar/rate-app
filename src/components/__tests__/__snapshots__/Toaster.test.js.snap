// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster rendering renders correctly 1`] = `ShallowWrapper {}`;
